### PR TITLE
chore: add a 'backend mock server' to ease frontend development

### LINKED ISSUE #18: [TEST] Add a way to simulate the Python server to ease frontend developments
Currently, when developing the front part of the demo, the Python server must be up.
This requires that "frontend" contributors setup it which is not very convenient (Java must also be configured).

To simplify the development process, I suggest we develop a fake JavaScript server that will replicate the Python server behavior.
If needed, this will also let easily test some edge cases: API errors, slow or fast responses, ....

The server has to replicate 2 API:
- XES to BPMN: discover/inductive-miner
- get frequency data: stats/frequency
- conformance alignment data file transformation: conformance/alignment
- conversion xes to csv : conversion/xes-to-csv

We could use [Express](https://expressjs.com/), [Fastify](https://www.fastify.io/), [NestJS](https://nestjs.com/), [Next.js](https://nextjs.org/), or others.

### Notes

This will help investigating #15 for instance

### PR DESCRIPTION
**Pull request by tbouffard**
This mock server provides the same API as the real backend. It is powered by the `NestJS` framework.
All http requests are logged thanks to the `pino-http` library.
There is no need to have XES local file when using the mock server: it returns arbitrary data.

Notes: the following APIs are available, but they need to be improved to allow a correct display in the frontend:
   - conversion/xes-to-csv
  - conformance/alignment

closes #18

### Screenshots

mock_server

### Notes

NestJS: https://nestjs.com/
source inspiration for the pino configuration: https://getpino.io/#/ and https://www.tomray.dev/nestjs-logging

Improvements that could follow
- #20
- #21
- #22
- #23

### CODE REVIEW

File: backend-mock-server/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# backend-mock-server
+
+Simulate the Python backend to ease the frontend developments.
```
Comment (tbouffard) [inline backend-mock-server/README.md:3]: **suggestion:** add link to the backend
```suggestion
Simulate the [Python backend](../backend/README.md) to ease the frontend developments.
```

File: backend-mock-server/src/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// declare module "template.html!text" {
```
Comment (tbouffard) [inline backend-mock-server/src/types.d.ts:1]: **nitpick:** Remove this file
It was an attempt to load the diagram from the assets. To be managed in #20

File: backend-mock-server/src/assets/diagram.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
+// TODO load assets from diagram.bpmn
+// import diagramContent from './diagram.bpmn!text';
+// export const diagram = diagramContent;
```
Comment (tbouffard) [inline backend-mock-server/src/assets/diagram.ts:3]: **suggestion:** remove extra comments
```suggestion
```

**Review by csouchet — APPROVED** [review]

File: backend-mock-server/src/types.d.ts (hunk 2 above)
Comment (tbouffard) [reply backend-mock-server/src/types.d.ts]: done in df772fc1
